cfg.y

```diff
@@ -1540,7 +1545,7 @@ assign_stm: LOGLEVEL EQUAL snumber { IFOR();
 		  }
 		| MCAST_TTL EQUAL error { yyerror("number expected as tos"); }
 		| TOS EQUAL NUMBER { IFOR(); tos = $3;
-							if (tos<=0)
+							if (tos<0)
```

Comment:
Why trashing the 0 value? Not sure if 0 is a legitimate value for TOS, but the current code allows you to set it.

Reply:
Thank you for reviewing. 0 here means "not set TOS".

Reply:
you say `setsockopt()` fails with a 0 TOS value ?

Reply:
By default, OpenSIPS sets the TOS value globally on all interfaces, and this behavior cannot be disabled.

This patch allows a TOS value of 0 to prevent TOS from being set at all.
In other words, if the global option tos=0 is configured, OpenSIPS will not apply TOS to the socket.

This change helps avoid validation issues when parsing the OpenSIPS script.

Reply:
Yes, exactly. By default, OpenSIPS sets TOS, which is not always the desired behavior. Setting TOS=0 disables it. We should have the option to avoid setting TOS both globally and on specific sockets.